src/arm/debugger/debugger.c

```diff
@@ -320,12 +435,22 @@ static void ARMDebuggerTrace(struct mDebuggerPlatform* d, char* out, size_t* len
 		}
 	}
 
-	*length = snprintf(out, *length, "%08X %08X %08X %08X %08X %08X %08X %08X %08X %08X %08X %08X %08X %08X %08X %08X cpsr: %08X | %s",
-		               cpu->gprs[0],  cpu->gprs[1],  cpu->gprs[2],  cpu->gprs[3],
-		               cpu->gprs[4],  cpu->gprs[5],  cpu->gprs[6],  cpu->gprs[7],
-		               cpu->gprs[8],  cpu->gprs[9],  cpu->gprs[10], cpu->gprs[11],
-		               cpu->gprs[12], cpu->gprs[13], cpu->gprs[14], cpu->gprs[15],
-		               cpu->cpsr.packed, disassembly);
+	size_t regStringLen = *length;
+	ARMDebuggerFormatRegisters(&cpu->regs, out, &regStringLen);
+	*length = regStringLen + snprintf(out + regStringLen, *length - regStringLen, " | %s", disassembly);
+}
+
+static void ARMDebuggerFormatRegisters(struct ARMRegisterFile* regs, char* out, size_t* length) {
+	*length = snprintf(out, *length, "%08X %08X %08X %08X %08X %08X %08X %08X %08X %08X %08X %08X %08X %08X %08X %08X cpsr: %08X",
+		               regs->gprs[0],  regs->gprs[1],  regs->gprs[2],  regs->gprs[3],
+		               regs->gprs[4],  regs->gprs[5],  regs->gprs[6],  regs->gprs[7],
+		               regs->gprs[8],  regs->gprs[9],  regs->gprs[10], regs->gprs[11],
+		               regs->gprs[12], regs->gprs[13], regs->gprs[14], regs->gprs[15],
+		               regs->cpsr.packed);
+}
+
+static void ARMDebuggerFrameFormatRegisters(struct mStackFrame* frame, char* out, size_t* length) {
+	ARMDebuggerFormatRegisters((struct ARMRegisterFile*)frame->regs, out, length);
```

Comment:
Casting from `void*` is implicit in C, so this cast isn't needed. (And if it were, mGBA style says there should be a space between the cast and the variable.)